eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -340,7 +340,16 @@ object InteractiveTxBuilder {
   }
   // @formatter:on
 
-  def apply(sessionId: ByteVector32,
+  // @formatter:off
+  sealed trait SessionContext {
+    def sessionId: ByteVector32
+  }
+  object SessionContext {
+    case class Unspecified(sessionId: ByteVector32) extends SessionContext
```

Comment:
For example, we could have a `LiquidityRequest` context which contains the requested `LiquidityAds.Lease`. There are various other cases in feature branches.